### PR TITLE
Updated the user guide to clarify how to save simulations results

### PR DESCRIPTION
**Pull request by dr-nate**
and progress. Also added a small note about the determinism of
simulations on different platforms.

### CODE REVIEW

**Comment by dr-nate** [comment]
Adds point from https://github.com/pandegroup/openmm/issues/1269 to the user guide.

File: docs-source/usersguide/application.rst [hunk 1]
```diff
@@ -1329,6 +1329,60 @@ separated by spaces instead of commas:
...
+like mdtraj or parmed). If you are simply interested in saving the structure,
+you can write it out as a PDB file using :code:`PDBFile.writeFile()`.  You can
+see an example of this in the modeller section :ref:`saving-the-results`.
+
+If you are hoping to save more information than just positions, you can use
+:code:`simulation.saveState()`. This will save the entire state of the
+simulation, including positions, velocities, box dimensions and much more in an
+XML file. This same file can be loaded back into OpenMM and used to continue
+the simulation. Importantly, because this file is a text file, it can be
+transfered between different platforms. A potential downside to this approach
```
Comment (swails) [inline docs-source/usersguide/application.rst]: transferred between different platforms *and* different versions of OpenMM, if that is desired.

File: docs-source/usersguide/application.rst [hunk 2]
```diff
@@ -1329,6 +1329,60 @@ separated by spaces instead of commas:
...
+
+    simulation.loadState('output.xml')
+
+There is a third way to save your simulation, known as a checkpoint file, which
+will save the entire simulation as a binary file. It will allow you to exactly
+continue a simulation if the need arises (though whether the simulation is
+deterministic depends on platform and methods, see
+:ref:`platform-specific-properties-determinism`). There are important caveats
+to this approach, however. This binary can only be used to restart simulations
+on machines with the same hardware and software as the one that saved it.
```
Comment (swails) [inline docs-source/usersguide/application.rst]: emphasize the same platform and same version of OpenMM as well.

**Comment by swails** [comment]
This PR has been open for awhile, but I think it's a nice addition to the manual.  I made a couple comments, but I think this is worth merging.

**Comment by peastman** [comment]
Agreed.  And your suggested changes make sense.  @dr-nate do you want to make those?

**Comment by dr-nate** [comment]
Yup, updated. Thanks @swails!

**Comment by swails** [comment]
Looks good to me

**Pull request merged**